rust/agents/relayer/src/relayer.rs

```diff
@@ -98,34 +95,55 @@ impl BaseAgent for Relayer {
             self.mailbox(&self.origin_chain).unwrap().db().clone(),
         ));
 
+        let prover_sync = Arc::new(RwLock::new(MerkleTreeBuilder::new(
+            self.mailbox(&self.origin_chain).unwrap().db().clone(),
+        )));
+        let mut send_channels: HashMap<u32, UnboundedSender<SubmitMessageArgs>> = HashMap::new();
+
         for chain in self.core.mailboxes.keys() {
             if *chain == self.origin_chain {
                 continue;
             }
+            let channels: (
```

Comment:
nit: definitely a personal bias of mine but I generally find non-deconstructed tuples pretty hard to read. How would you feel about something like this?

```
let (message_sender, message_receiver) = ...
```

Reply:
Yes for sure that's better, still learning rust :)